File: INTRODUCCION-POO/EJ_9/EJ_9.py
class Computadora:
    def __init__(self, procesador, ram, almacenamiento):
        self.procesador = procesador
        self.ram = ram
        self.almacenamiento = almacenamiento
        self.encendida = False

    def mostrar_componentes(self):
        print(f"Procesador: {self.procesador}")
        print(f"RAM: {self.ram}")
        print(f"Almacenamiento: {self.almacenamiento}")

    def encender(self):
        if not self.encendida:
            self.encendida = True
            print("Computadora encendida.")
        else:
            print("Ya estaba encendida.")

    def apagar(self):
        if self.encendida:
            self.encendida = False
            print("Computadora apagada.")
        else:
            print("Ya estaba apagada.")

    def estado(self):
        print("Estado:", "Encendida" if self.encendida else "Apagada")


pc = Computadora("Intel i7", "16GB", "1TB SSD")
pc.mostrar_componentes()
pc.estado()
pc.encender()
pc.estado()
pc.apagar()
pc.estado()

File: INTRODUCCION-POO/EJ_9/EJ_9.java
public class Computadora {
    String procesador;
    String ram;
    String almacenamiento;
    boolean encendida;

    public Computadora(String procesador, String ram, String almacenamiento) {
        this.procesador = procesador;
        this.ram = ram;
        this.almacenamiento = almacenamiento;
        this.encendida = false;
    }

    public void mostrarComponentes() {
        System.out.println("Procesador: " + procesador);
        System.out.println("RAM: " + ram);
        System.out.println("Almacenamiento: " + almacenamiento);
    }

    public void encender() {
        if (!encendida) {
            encendida = true;
            System.out.println("Computadora encendida.");
        } else {
            System.out.println("Ya estaba encendida.");
        }
    }

    public void apagar() {
        if (encendida) {
            encendida = false;
            System.out.println("Computadora apagada.");
        } else {
            System.out.println("Ya estaba apagada.");
        }
    }

    public void estado() {
        System.out.println("Estado: " + (encendida ? "Encendida" : "Apagada"));
    }

    public static void main(String[] args) {
        Computadora pc = new Computadora("Intel i7", "16GB", "1TB SSD");
        pc.mostrarComponentes();
        pc.estado();
        pc.encender();
        pc.estado();
        pc.apagar();
        pc.estado();
    }
}
